src/lib/components/activity/activity-options-update-drawer.svelte

```diff
@@ -0,0 +1,304 @@
+  import type { PendingActivity } from '$lib/types/events';
+  import { formatSecondsAbbreviated } from '$lib/utilities/format-time';
+
+  type Props = {
+    open: boolean;
+    namespace: string;
+    execution: WorkflowExecution;
+    activity: PendingActivity;
+  };
+
+  const fromDurationToNumber = (duration) => {
+    if (!duration || !duration.endsWith('s')) {
+      return '';
+    }
+
+    return duration?.replace('s', '');
+  };
+
+  const fromNumberToDuration = (duration: number): string => {
+    if (!duration) return undefined;
+    return duration + 's';
+  };
+
+  let { open = $bindable(), namespace, execution, activity }: Props = $props();
+  let { activityId: id, activityType: type } = $derived(activity);
+  let taskQueue = $state(activity.activityOptions?.taskQueue?.name);
+  let scheduleToCloseTimeout = $state(
+    fromDurationToNumber(activity?.activityOptions?.scheduleToCloseTimeout),
+  );
+  let scheduleToStartTimeout = $state(
+    fromDurationToNumber(activity?.activityOptions?.scheduleToStartTimeout),
+  );
+  let startToCloseTimeout = $state(
+    fromDurationToNumber(activity?.activityOptions?.startToCloseTimeout),
+  );
+  let heartbeatTimeout = $state(
+    fromDurationToNumber(activity?.activityOptions?.heartbeatTimeout),
+  );
+  let maximumAttempts = $state(
+    activity?.activityOptions?.retryPolicy?.maximumAttempts,
+  );
+  let backoffCoefficient = $state(
+    activity?.activityOptions?.retryPolicy?.backoffCoefficient,
+  );
+  let initialInterval = $state(
+    fromDurationToNumber(
+      activity?.activityOptions?.retryPolicy?.initialInterval,
+    ),
+  );
+  let maximumInterval = $state(
+    fromDurationToNumber(
+      activity?.activityOptions?.retryPolicy?.maximumInterval,
+    ),
+  );
+  let includeType = $state(false);
+
+  const activityOptions = $derived({
+    taskQueue: { name: taskQueue },
+    scheduleToCloseTimeout: fromNumberToDuration(scheduleToCloseTimeout),
+    scheduleToStartTimeout: fromNumberToDuration(scheduleToStartTimeout),
+    startToCloseTimeout: fromNumberToDuration(startToCloseTimeout),
+    heartbeatTimeout: fromNumberToDuration(heartbeatTimeout),
+    retryPolicy: {
+      maximumAttempts,
+      initialInterval: fromNumberToDuration(initialInterval),
+      backoffCoefficient,
+      maximumInterval: fromNumberToDuration(maximumInterval),
+    },
+  }) as unknown as ActivityOptions;
+
+  const closeCustomizationDrawer = () => {
+    open = false;
+  };
+
+  const onUpdate = async () => {
+    try {
+      await updateActivityOptions({
+        namespace,
+        execution,
+        id: includeType ? undefined : id,
+        type: includeType ? type : undefined,
+        activityOptions,
+      });
+      $refresh = Date.now();
+      toaster.push({
+        variant: 'success',
+        message: `Options for Activity ${id} have been updated.`,
+      });
+    } catch (error) {
+      console.error('Error updating activity options:', error);
```

Comment:
Do we want to let this error throw so its caught in an error boundary? Or show a toast to the user? Just swallowing seems odd.